crypto/cloudkms/cloudkms_test.go

```diff
@@ -45,3 +51,84 @@ func TestKeyFromResourceID(t *testing.T) {
 
 	assert.Equal(t, key, keyFromResourceID)
 }
+
+// gcloudApplicationSignin signs in as an application user using gcloud command line tool
+// currently assumes gcloud is already installed on the machine
+// will by default pop a browser window to sign in
+func gcloudApplicationSignin(kms cloudkms.Key) error {
+	googleAppCreds := os.Getenv("GOOGLE_APPLICATION_CREDENTIALS")
+	if len(googleAppCreds) > 0 {
+		return nil
+	}
+
+	proj := kms.ProjectID
+	if len(proj) == 0 {
+		return fmt.Errorf(
+			"could not get GOOGLE_APPLICATION_CREDENTIALS, no google service account JSON provided but private key type is KMS",
+		)
+	}
+
+	loginCmd := exec.Command("gcloud", "auth", "application-default", "login", fmt.Sprintf("--project=%s", proj))
+
+	output, err := loginCmd.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("Failed to run %q: %s\n", loginCmd.String(), err)
+	}
+
+	squareBracketRegex := regexp.MustCompile(`(?s)\[(.*)\]`)
+	regexResult := squareBracketRegex.FindAllStringSubmatch(string(output), -1)
+	// Should only be one value. Second index since first index contains the square brackets
+	googleApplicationCreds := regexResult[0][1]
+
+	os.Setenv("GOOGLE_APPLICATION_CREDENTIALS", googleApplicationCreds)
+
+	return nil
+}
+
+// TestManualKMSSigning tests signing using a KMS key.
+// This tests requires access to KMS and cannot be run by CI.
+// Please use this test manually by commenting t.Skip(),
+// when making any change to the KMS signing code.
+func TestManualKMSSigning(t *testing.T) {
+	// to comment when testing manually
+	t.Skip()
+
+	key := cloudkms.Key{
+		ProjectID:  "dl-flow",
+		LocationID: "us-east1",
+		KeyRingID:  "testnet_keyring",
+		KeyID:      "sdk_test",
+		KeyVersion: "1",
+	}
+
+	// id is copied from the kms key: https://console.cloud.google.com/security/kms/key/manage/us-east1/testnet_keyring/sdk_test?project=dl-flow
+	// this test insures the key exists in the kms keyring
+	id := "projects/dl-flow/locations/us-east1/keyRings/testnet_keyring/cryptoKeys/sdk_test/cryptoKeyVersions/1"
```

Comment:
Can we make this an env var, like `GOOGLE_APPLICATION_CREDENTIALS`? I don't feel great about putting GCP details in code, even if they aren't technically secrets.

```go
id := os.Getenv(`KEY_RESOURCE_ID`)
key := cloudkms.KeyFromResourceID()
```

I'd then suggest we use a separate test to check that `key.ResourceID()` behaves as expected.

Reply:
Nice, I like your suggestion, added it in https://github.com/onflow/flow-go-sdk/pull/283/commits/cb01b7dc5629c7f3f6b3681c82631c7a4c01747d